### ISSUE TITLE
Food item loses nutrition after cooked.

### ISSUE THREAD

**Issue by AndrXi** [CleverRaven/Cataclysm-DDA#27911]
**Describe the bug**  
One of the food items has no calorie and vitamin value after cooked.
![bug report 1](https://user-images.githubusercontent.com/44873249/51793736-2e401d00-21bd-11e9-902e-a169f42819c9.png)
![bug report 2](https://user-images.githubusercontent.com/44873249/51793737-3009e080-21bd-11e9-91e2-20e93ea5ce15.png)

**To Reproduce**  
1. Try to make vegetable aspic.
2. Check result.

**Expected behavior**  
Display the correct values.

**Versions and configuration(please complete the following information):**
 - OS: Win 7
 - Game Version: 0.C-37094-g5da8e39 (tiles)
 - **Mods loaded:**  "dda",  "no_npc_food",  "filthy_morale",  "ew_pack",  "makeshift",  "Medieval_Stuff",
  "More_Survival_Tools",  "modular_turrets",  "Salvaged_Robots",  "alt_map_key",  "Mining_Mod",
  "mutant_npcs",  "npc_traits",  "more_locations",  "FujiStruct",  "boats",  "deoxymod",  "blazemod",
  "Tanks",  "DP_REMIX_INDICATORS",  "StatsThroughSkills",  "PK_REBALANCE",  "growable-pots"

**Comment by moxian** [comment]
I think the solution should be to give bones nutritional value, but make them unconsumable, the way flour is, rather than hardcode aspic calories directly.
That way both aspic and bone broth would get the nutrition from the ingredients "naturally".

**Comment by moxian** [comment]
Separately, I think it's impossible in real life to make aspic without meat/bones. Should maybe bones be moved to an ingredient of their own, instead of being replaceable by vegetable matter as it is now?

**Comment by Dragometh** [comment]
> Separately, I think it's impossible in real life to make aspic without meat/bones. Should maybe bones be moved to an ingredient of their own, instead of being replaceable by vegetable matter as it is now?

https://ukrainian-recipes.com/vegetable-aspic.html
Apparently, it is possible.



> I think the solution should be to give bones nutritional value, but make them unconsumable, the way flour is, rather than hardcode aspic calories directly.
> That way both aspic and bone broth would get the nutrition from the ingredients "naturally".

The issue here is that the calories value is either not being displayed or is being completely lost after the item is created.
@AndrXi When you eat it, does it reduce hunger, or it doesn't change anything?

**Comment by KorGgenT** [comment]
i saw this bug. i've got a handle on it. PR is incoming soon.

**Comment by KorGgenT** [comment]
in the meantime: bones have no nutritional value. neither does water 0+0=0
therefore it doesn't technically lose nutrition, as it never had it in the first place.

**Comment by FuelType-Memes** [comment]
> 
> 
> in the meantime: bones have no nutritional value. neither does water 0+0=0
> therefore it doesn't technically lose nutrition, as it never had it in the first place.

Maybe change the recipe to have more components like the meat aspic, but bones + veggie? 

Because right now, for example, veg aspic made with wild vegetables is higher lvl than just cooked wild vegetables, has additional water requirement, takes more time and has 0 benefits

**Comment by Dragometh** [comment]
Don't know if related, but upon checking veggie aspic recipe code and re-checking the recipe image, I found this line:

`"components": [ [ [ "bone", 4 ], [ "veggy_any", 2, "LIST" ] ], [ [ "water_clean", 2 ], [ "water", 2 ] ] ]`´

In other words, you have an option of **either** using vegetables or using bones. Is this working as intended?

**Comment by Dragometh** [comment]
> in the meantime: bones have no nutritional value. neither does water 0+0=0
> therefore it doesn't technically lose nutrition, as it never had it in the first place.

I didn't know they added that change to the code. Found it out after checking recipe and food code.

> Should maybe bones be moved to an ingredient of their own, instead of being replaceable by vegetable matter as it is now?

Don't know if this is working as intended, but on checking the recipe code, there is this line:

`"//": "Gelatin from bones is just boiling them long enough.  Gelatin from pectin is harder.  Instead of duping the ingredients, *1.5 them for aforementioned pectin.",`

Now, if this is being applied in the code, I don't know. The ingredients line is:

`"components": [ [ [ "bone", 4 ], [ "veggy_any", 2, "LIST" ] ], [ [ "water_clean", 2 ], [ "water", 2 ] ] ]`

Quick google search led me to finding out that pectin is made from fruit and acid (lemon is the most common, I believe, but there probally are other options). If so, then pectin should be an separate ingredient, making the veggie aspic recipe require water, pectin or bones and vegetable.

> I think the solution should be to give bones nutritional value, but make them unconsumable, the way flour is, rather than hardcode aspic calories directly.
> That way both aspic and bone broth would get the nutrition from the ingredients "naturally".

I also agree on this, mostly because in real life bones also do have nutrition value, no?
The problem is finding out how much is the nutritional value of it.

**Comment by Dragometh** [comment]
On another side-note, should bones be considered together as vegetable and be edible by characters with the Meat Intolerant trait?

**Comment by SirPendrak** [comment]
This is very similar isue, so i wont be opening new thread, and write it here:
Raw macaroni has 0 calories. If you cook it you will get 0 calories boiled nuddles. 
Boiled noodles from raw lasagne or spaghetti has 217 valories, and from fast noodles - ~300 calories.
It seems strange also that your body can get all the nutrition from raw uncooked lasagne, it should be different values, just like in meat case.

**Comment by moxian** [comment]
> > Separately, I think it's impossible in real life to make aspic without meat/bones. Should maybe bones be moved to an ingredient of their own, instead of being replaceable by vegetable matter as it is now?
> 
> https://ukrainian-recipes.com/vegetable-aspic.html
> Apparently, it is possible.

The recipe you've linked has store-bought gelatin as an ingredient. Gelatin comes from bones. I've tried searching the internet for more examples, but they all have gelatin as one of the ingredients. I don't think this counts as a proof that it's possible.

> Because right now, for example, veg aspic made with wild vegetables is higher lvl than just cooked wild vegetables, has additional water requirement, takes more time and has 0 benefits

If I'm not mistaken the primary historical use for aspic is food preservation, since it prevents the access of oxygen and bacteria to the matter within (well, that, and extraction of nutrients from the bones). Right now both veggies and aspic spoil in 3.25 days. Would increasing the shelf time of aspic make sense?

> Don't know if this is working as intended, but on checking the recipe code, there is this line:
 >`"//": "Gelatin from bones is just boiling them long enough. Gelatin from pectin is harder. Instead of duping the ingredients, *1.5 them for aforementioned pectin.",`
> Quick google search led me to finding out that pectin is made from fruit and acid (lemon is the most common, I believe, but there probally are other options). If so, then pectin should be an separate ingredient, making the veggie aspic recipe require water, pectin or bones and vegetable.

There's not much pectin in vegetables, no. If you substitute vegetables for fruit, at best consistency that you'd get is that of jam - and then you've just made jam. Sure you can put other vegetables into jam, but that sounds neither tasty nor particularly useful (and it definitely won't be aspic).

> Quick google search led me to finding out that pectin is made from fruit and acid (lemon is the most common, I believe, but there probally are other options). If so, then pectin should be an separate ingredient, making the veggie aspic recipe require water, pectin or bones and vegetable.

My google search says it's most common to make pectin at home from apples or oranges - no extra acid lemons needed.
However replacing bones with pectin in aspic does not sound like a good idea - I haven't found any recipes that substitute gelatin/bones with pectin online other than a [single one by pectin production company](http://www.pomonapectin.com/recipes/tomato-shrimp-aspic/).
I would guess that's because pectin requires sugar to become gel, and sweet aspic is not particularly tasty? Or maybe because it has somewhat less structural integrity than aspic (does that even matter?)..
But probably the most likely reason is that if you use pectin, you'd get jelly instead of aspic. I'm totally fine having non-royal jellies in the game (but that's probably a separate effort).

>The problem is finding out how much is the nutritional value of [bones].

We can infer the value from the bone broth caloric contents. [This recipe](https://recipes.sparkpeople.com/recipe-calories.asp?recipe=2041648) says there are 66 calories per 56.75 gram of bones. Aspic will definitely be more nutritious, since bones are cooked for longer and more good stuff is pulled out of them, but using the value for bone broth there as a first approximation sounds alright to me.
It might be a good idea to show what percentage of nutrients get transferred from the ingredient to the resulting product, but that sounds very complicated to implement and long-term.


**TL;DR:** 
- no, you can't replace bones in vegetable aspic with fruit/vegetables - make jellies from those instead
- aspic should have longer shelf life
- bones nutritional value can be (under-)estimated at 116 calories per 100 gram

**Comment by Barhandar** [comment]
The aspic is supposed to not spoil, and has not before the rather misguided comestible spoilage overhaul. There are separate recipes for regular and vegetable aspic; **it's entirely intended to NOT take bones**, @KorGgenT.

**Comment by moxian** [comment]
You can't make vegetable aspic without bones, and still call it vegetable aspic. Just as you can't make it without vegetables.

I don't understand why you want vegetable aspic *specifically* being made with no animal-sourced ingredients - there should be plenty of no-animal recipes to choose an alternative from. If it's an attempt to go vegetarian, I don't think the base game should compromise realism for that (that stuff should go into a mod).

If you really want aspic-like thing without bones - you are free to make (as in, introduce into the game) jelly recipe, replacing bones with fruit as a binding agent. However that still won't be called "vegetable aspic", but "vegetable jelly" instead.

**Comment by KorGgenT** [comment]
just letting you all know that what i did was separate the bones and veggies -- requiring both.

**Comment by Barhandar** [comment]
>You can't make vegetable aspic without bones, and still call it vegetable aspic. Just as you can't make it without vegetables.

You can make vegetable aspic with agar or carrageenan (admittedly those need ocean to be added to the game). For piscetarians, some fish can be used to make isinglass, which also works as gelatin.
